db_stress_tool/db_stress_test_base.cc

```diff
@@ -2423,6 +2427,80 @@ Status StressTest::TestGetPropertiesOfAllTables() const {
   return db_->GetPropertiesOfAllTables(&props);
 }
 
+void StressTest::TestGetUpdatesSince(ThreadState* thread) {
+  Status s = db_->DisableFileDeletions();
+  if (!s.ok()) {
+    return;
+  }
+
+  const SequenceNumber since_seq =
+      thread->rand.Next64() % (db_->GetLatestSequenceNumber() + 1);
+  std::unique_ptr<TransactionLogIterator> iter;
+
+  s = db_->GetUpdatesSince(since_seq, &iter);
+
+  if (!s.ok()) {
+    fprintf(stderr,
+            "Failed to call GetUpdatesSince(): %s with specified "
+            "seq num: %" PRIu64 "\n",
+            s.ToString().c_str(), since_seq);
+    thread->shared->SafeTerminate();
+  }
+
+  if (!iter->status().ok()) {
+    fprintf(stderr,
+            "Iterator with non-ok status returned by "
+            "GetUpdatesSince(): %s \n",
+            iter->status().ToString().c_str());
+    thread->shared->SafeTerminate();
+  } else if (!iter->Valid()) {
+    return;
+  }
+
+  BatchResult res = iter->GetBatch();
+  SequenceNumber last_seqno = res.sequence;
+  iter->Next();
+
+  while (iter->status().ok() && iter->Valid()) {
+    res = iter->GetBatch();
+    if (res.sequence <= last_seqno) {
+      fprintf(stderr,
+              "Found a batch result of non-increasing seq num %" PRIu64
+              " from previous scan. Previous batch of seq num: %" PRIu64
+              ", current batch of seq num: %" PRIu64
+              ". The iterator used was returned from GetUpdatesSince() with "
+              "specified "
+              "seq num: %" PRIu64 "\n",
+              res.sequence, last_seqno, res.sequence, since_seq);
+      thread->shared->SafeTerminate();
+    }
+    last_seqno = res.sequence;
+    iter->Next();
+  }
+
+  // `TryAgain()` is allowed as it might happens when new writes happen between
+  // calling `GetLatestSequenceNumber()` above and `iter->Next()`
```

Comment:
Sorry I don't understand. `GetLatestSequenceNumber()` appears to be the maximum possible value of the lower bound we pass to `GetUpdatesSince()`.